From: Shiraz Saleem <shirazsaleem@microsoft.com>
Date: Wed, 14 Aug 2024 14:40:39 -0700
Subject: providers/mana: Fix return value on unsupported QP type

Return EOPNOTSUPP when QP type is not supported during create QP.

Fixes: 443f196deee0 ("mana: Microsoft Azure Network Adapter (MANA) RDMA provider")
Signed-off-by: Shiraz Saleem <shirazsaleem@microsoft.com>
Origin: upstream, https://github.com/linux-rdma/rdma-core/pull/1556
---
 providers/mana/qp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/providers/mana/qp.c b/providers/mana/qp.c
index 596ab75..a6b28d5 100644
--- a/providers/mana/qp.c
+++ b/providers/mana/qp.c
@@ -327,7 +327,7 @@ struct ibv_qp *mana_create_qp(struct ibv_pd *ibpd,
 	default:
 		verbs_err(verbs_get_ctx(ibpd->context),
 			  "QP type %u is not supported\n", attr->qp_type);
-		errno = EINVAL;
+		errno = EOPNOTSUPP;
 	}
 
 	return NULL;
@@ -533,7 +533,7 @@ struct ibv_qp *mana_create_qp_ex(struct ibv_context *context,
 	default:
 		verbs_err(verbs_get_ctx(context),
 			  "QP type %u is not supported\n", attr->qp_type);
-		errno = EINVAL;
+		errno = EOPNOTSUPP;
 	}
 
 	return NULL;
